scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUri.scala

```diff
@@ -109,15 +109,24 @@ private[sparkey] class SparkeyWriter(val uri: SparkeyUri) {
       case gcsUri: GcsSparkeyUri => {
         // Copy .spi and .spl to GCS path
         for (ext <- Seq("spi", "spl")) {
-          val writer = gcsUri.gcs.create(
+          val gcsFile = gcsUri.gcs.create(
             GcsPath.fromUri(s"${gcsUri.basePath}.$ext"),
             "application/octet-stream")
-          writer.write(ByteBuffer.wrap(Files.readAllBytes(Paths.get(s"$localFile.$ext"))))
-          writer.close()
+          val localFileChannel = FileChannel.open(Paths.get(s"$localFile.$ext"))
+          transferFile(localFileChannel, gcsFile)
         }
       }
       case _ => ()
     }
   }
 
+  private def transferFile(srcChannel: FileChannel, destination: WritableByteChannel): Unit = {
+    var position = 0L
+    val srcFileSize = srcChannel.size()
+    while (position < srcFileSize) {
+      position += srcChannel.transferTo(position, srcFileSize - position, destination)
```

Comment:
Googling for how this function works, it is hard to find an answer on whether this actually buffers the transfer for you (or tries to read the whole file into memory. Will need to investigate this tomorrow.